query/src/main/scala/filodb/query/exec/SetOperatorExec.scala

```diff
@@ -97,25 +97,47 @@ final case class SetOperatorExec(queryContext: QueryContext,
     else rv.rows.filter(!_.getDouble(1).isNaN).isEmpty
   }
 
+  // scalastyle:off method.length
   private def setOpAnd(lhsRvs: List[RangeVector], rhsRvs: List[RangeVector],
-                       rhsSchema: ResultSchema): List[RangeVector] = {
+                       rhsSchema: ResultSchema): Iterable[RangeVector] = {
     // isEmpty method consumes rhs range vector
     require(rhsRvs.forall(_.isInstanceOf[SerializedRangeVector]), "RHS should be SerializedRangeVector")
+
+    val result = new mutable.HashMap[Map[Utf8Str, Utf8Str], ArrayBuffer[RangeVector]]()
     val rhsMap = new mutable.HashMap[Map[Utf8Str, Utf8Str], RangeVector]()
-    var result = new ListBuffer[RangeVector]()
+
     rhsRvs.foreach { rv =>
       val jk = joinKeys(rv.key)
       // Don't add range vector if it is empty
-      if (jk.nonEmpty && !isEmpty(rv, rhsSchema))
-        rhsMap.put(jk, rv)
+      if (jk.nonEmpty && !isEmpty(rv, rhsSchema)) {
+        if (rhsMap.contains(jk)) {
+          val resVal = rhsMap(jk)
+          if (resVal.key.labelValues == rv.key.labelValues) {
+            rhsMap.put(jk, StitchRvsExec.stitch(rv, resVal) )
```

Comment:
Lets document why we are stitching.. "When spread changes, we need to account for multiple Range Vectors with same key coming from different shards. Each of these range vectors would contain data for different time ranges". I missed adding this piece of information in the BinaryJoinExec as well clearly. Do you mind adding there too ?